### PR TITLE
Heur brycor

### PR DESCRIPTION
**Pull request by awig**
Also includes method for discrete tap stepping (rounding at each iteration)

### CODE REVIEW

File: gridopt/power_flow/ac_pf.py [hunk 1]
```diff
@@ -558,16 +569,28 @@ def c2(s):
...
 
         def c3(s):
+            """PV-PQ start at given iteration"""
             if (s.k >= params['pvpq_start_k'] and params['Q_limits'] and Q_mode == self.CONTROL_MODE_REG):
                 prob = s.problem.wrapped_problem
                 prob.apply_heuristics(s.x)
                 s.problem.A = prob.A
                 s.problem.b = prob.b
 
+        def c4(s):
```
Comment (fxz89) [inline gridopt/power_flow/ac_pf.py:579]: It seems the prob.apply_heuristics(s.x) in different callbacks will interfere with each others. For example my transformer P Q control heuristics, even without adding new callback, they will be called by c3; adding a callback for transformer P, Q control and not have the same results. Not sure if we want to further specify the heuristics to be applied.

Comment (awig) [reply gridopt/power_flow/ac_pf.py:579]: Agreed. We should pass the apply_heuristics to pass the specific heuristic to apply.

Comment (awig) [reply gridopt/power_flow/ac_pf.py:680]: Basee on Tamer's latest changes, the heuristic is no longer needed and we only need the new constraint. I'll test it to make sure but this now avoids this problem for this case at least.

**Comment by awig** [comment]
Updating to use the constraints and variables that @TamerSobhy added and will delete the heuristic that I created.

**Comment by awig** [comment]
Updated with y-scale/correction variables and constraint.

**Pull request merged**